sql/src/main/java/io/crate/execution/jobs/PKLookupContext.java

```diff
@@ -59,29 +59,30 @@
     private final RowConsumer consumer;
     private final InputRow inputRow;
     private final List<CollectExpression<GetResponse, ?>> expressions;
+    private final PKLookupPhase phase;
 
-    public PKLookupContext(UUID jobId,
-                           int phaseId,
+    public PKLookupContext(PKLookupPhase phase,
```

Comment:
I'm wondering if we're optimizing for less-typing here instead of making things more readable. By passing in the whole phase we're loosing the information that things like `projections` are *not* used by the context.

Reply:
I found it rather confusing that certain Context have their phase passed as an argument and some only the phaseId/name. I know it make sense to pass only properties that are really necessary, but that does not encourage consistency.
However, I'm ok with just passing in the name additionally.